core/to_from_v8.rs

```diff
@@ -17,3 +20,152 @@ pub trait FromV8<'a>: Sized {
     value: v8::Local<'a, v8::Value>,
   ) -> Result<Self, Self::Error>;
 }
+
+// impls
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
We're going to need to ensure the docs for this are decent. We can fill in more details later:

```suggestion
#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
/// Marks a numeric type as being serialized as a v8 `smi` in a `v8::Integer`.
```